src/modules/api/graphql/resolvers.ts

```diff
@@ -104,5 +143,32 @@ export const Resolvers = {
       }
     })
     return nextSlot;
+  },
+
+  // finds and tags vsc-tx/ vsc-blocks via ipfs CID's, unidentifiable CID's are tagged with the type 'null'
+  findCID: async (_, args) => {
+    if (appContainer.self.config.get('ipfs.pinEverything')) {
+      const ipfsHash = args.id;
+  
+      if (!ipfsHash) {
+        return null;
+      }
+  
+      if (ipfsHash.startsWith('Qm') || ipfsHash.startsWith('bafy')) {
```

Comment:
Is there anything else (other than contract manifest/code) that uses DAG-PD (0x70)? Everything else I've seen are using DAG-CBOR (0x71) that only begins with `bafyrei`.

Perhaps doing a `CID.parse()` in a try/catch then checking if code is 0x71 should be the first things to validate.

Reply:
not that im aware of atm

changed the code like you suggested, now it's way more readable!